installer/setup_base.txt

```diff
@@ -18,63 +18,82 @@ const config = {
       check: (message) => {
         try {
-          const modRole = message.guild.roles.find(r => r.name.toLowerCase() === message.settings.modRole.toLowerCase());
-          if (modRole && message.member.roles.has(modRole.id)) return true;
+          const modRole = message.guild.roles.cache.find(r => r.name.toLowerCase() === message.settings.modRole.toLowerCase());
+          if (modRole && message.member.roles.cache.has(modRole.id)) return true;
         } catch (e) {
           return false;
         }
       }
     },
 
-    {
-      level: 3,
+    { level: 3,
       name: "Administrator", 
       check: (message) => {
         try {
-          const adminRole = message.guild.roles.find(r => r.name.toLowerCase() === message.settings.adminRole.toLowerCase());
-          return (adminRole && message.member.roles.has(adminRole.id));
+          const adminRole = message.guild.roles.cache.find(r => r.name.toLowerCase() === message.settings.adminRole.toLowerCase());
+          return (adminRole && message.member.roles.cache.has(adminRole.id));
         } catch (e) {
           return false;
         }
       }
     },
-
-    {
-      level: 4,
+    // This is the server owner.
+    { level: 4,
       name: "Server Owner", 
-      check: (message) => message.channel.type === "text" ? (message.guild.owner.user.id === message.author.id ? true : false) : false
+      // Simple check, if the guild owner id matches the message author's ID, then it will return true.
+      // Otherwise it will return false.
+      check: (message) => message.channel.type === "text" ? (message.guild.ownerID === message.author.id ? true : false) : false
     },
 
-    {
-      level: 8,
+    // Bot Support is a special inbetween level that has the equivalent of server owner access
+    // to any server they joins, in order to help troubleshoot the bot on behalf of owners.
+    { level: 8,
       name: "Bot Support",
+      // The check is by reading if an ID is part of this array. Yes, this means you need to
+      // change this and reboot the bot to add a support user. Make it better yourself!
       check: (message) => config.support.includes(message.author.id)
     },
 
-    { 
-      level: 9,
+    // Bot Admin has some limited access like rebooting the bot or reloading commands.
+    { level: 9,
       name: "Bot Admin",
       check: (message) => config.admins.includes(message.author.id)
     },
 
-    { 
-      level: 10,
+    // This is the bot owner, this should be the highest permission level available.
+    // The reason this should be the highest level is because of dangerous commands such as eval
+    // or exec (if the owner has that).
+    { level: 10,
       name: "Bot Owner", 
-      check: (message) => message.client.owners.includes(message.author.id)
+      // Another simple check, compares the message author id to the one stored in the config file.
+      check: (message) => message.client.config.ownerID === message.author.id
```

Comment:
This does not take advantage of the teams that was implemented by Discord a little while ago. revert these changes.